driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/FindPublisherImpl.java

```diff
@@ -199,14 +200,15 @@
     private <E> Publisher<E> publishExplain(final Class<E> explainResultClass, @Nullable final ExplainVerbosity verbosity) {
         notNull("explainDocumentClass", explainResultClass);
         return getMongoOperationPublisher().createReadOperationMono(() ->
-                        asAsyncReadOperation().asAsyncExplainableOperation(verbosity,
+                        asAsyncReadOperation(0).asAsyncExplainableOperation(verbosity,
```

Comment:
The explain command doesn't use the batchSize but because of the change to push batchSize down we have to add a value and the server default seems best - as it may be honoured in the future.

`BatchCursorPublisher.asAsyncFirstReadOperation` only will return the first result so BatchSize can be set to 1.

Reply:
@rozza, thank you for the explanation, it makes all the sense! :)